### PR TITLE
Drive Widget details are now hidden when they are not available

### PR DESCRIPTION
**Pull request by d2dyno1**
BEFORE:
image

AFTER:
image

### CODE REVIEW

File: Files/UserControls/Widgets/DrivesWidget.xaml [hunk 1]
```diff
@@ -124,29 +124,41 @@
                                     FontFamily="{StaticResource FluentUIGlyphs}"
                                     FontSize="28"
                                     Glyph="{x:Bind Glyph}" />
-                                <StackPanel
+                                <Grid
                                     Grid.Column="1"
                                     Margin="12,0,0,0"
```
Comment (yaichenbaum) [inline Files/UserControls/Widgets/DrivesWidget.xaml:129]:
```suggestion
                                    Margin="12,0,0,0"
                                    RowSpacing="8"
```

File: Files/UserControls/Widgets/DrivesWidget.xaml [hunk 2]
```diff
@@ -124,29 +124,41 @@
...
+                                    
                                     <TextBlock
                                         x:Name="ItemLocationName"
                                         FontSize="14"
                                         FontWeight="Medium"
                                         Text="{x:Bind Text, Mode=OneWay}"
-                                        TextWrapping="NoWrap" />
+                                        TextWrapping="NoWrap"
+                                        VerticalAlignment="Center"
+                                        Margin="0 0 0 8"
```
Comment (yaichenbaum) [inline Files/UserControls/Widgets/DrivesWidget.xaml:144]:
```suggestion
```

File: Files/UserControls/Widgets/DrivesWidget.xaml [hunk 3]
```diff
@@ -124,29 +124,41 @@
...
+                                        Grid.Row="0"/>
                                     <muxc:ProgressBar
                                         AutomationProperties.AccessibilityView="Raw"
                                         Maximum="{x:Bind MaxSpace.GigaBytes, Mode=OneWay}"
-                                        Value="{x:Bind SpaceUsed.GigaBytes, Mode=OneWay}" />
+                                        Value="{x:Bind SpaceUsed.GigaBytes, Mode=OneWay}"
+                                        Visibility="{x:Bind ShowDriveDetails, Mode=OneWay}"
+                                        Margin="0 0 0 8"
```
Comment (yaichenbaum) [inline Files/UserControls/Widgets/DrivesWidget.xaml:151]:
```suggestion
```

**Review by yair100 — DISMISSED** [review]

**Comment by yaichenbaum** [comment]
:shipit:

**Comment by d2dyno1** [comment]
With `RowSpacing`, the text is not centered:

With `RowSpacing`:
image

Without `RowSpacing`:
image

@yair100

File: Files/UserControls/Widgets/DrivesWidget.xaml [hunk 4]
```diff
@@ -124,34 +124,45 @@
...
                                     Margin="12,0,0,0"
-                                    VerticalAlignment="Center"
-                                    Orientation="Vertical"
-                                    Spacing="8">
+                                    RowSpacing="8"
+                                    VerticalAlignment="Center">
+                                    <Grid.RowDefinitions>
+                                        <RowDefinition/>
+                                        <RowDefinition Height="Auto"/>
+                                        <RowDefinition Height="Auto"/>
```
Comment (yaichenbaum) [inline Files/UserControls/Widgets/DrivesWidget.xaml:135]:
```suggestion
                                        <RowDefinition Height="*"/>
                                        <RowDefinition Height="*"/>
                                        <RowDefinition Height="*"/>
```

Comment (d2dyno1) [reply Files/UserControls/Widgets/DrivesWidget.xaml:135]: It does not change anything with `RowSpacing`, I guess only `Margin` works in this case

Comment (yaichenbaum) [reply Files/UserControls/Widgets/DrivesWidget.xaml:135]: @d2dyno1 Is there a reason you changed the StackPanel to a Grid?

Comment (d2dyno1) [reply Files/UserControls/Widgets/DrivesWidget.xaml:135]: Hmmm, for some reason I though it wouldn't work with `StackPanel` lol. It works, but again if we want to have the name centered we have to ditch `Spacing` and replace it with `Margin` for every item

**Comment by d2dyno1** [comment]
Should I replace `Spacing` with `Margin`?

@yair100

**Comment by yair100** [comment]
> Should I replace `Spacing` with `Margin`?
> 
> @yair100

Yes.

**Comment by d2dyno1** [comment]
Merge issues 💁‍♂️

**Comment by d2dyno1** [comment]
:shipit:

**Review by yair100 — APPROVED** [review]

**Pull request merged**